apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -102,27 +125,34 @@ def stac_api_to_odc(
     else:
         logging.warning("API did not return the number of items.")
 
-    # Get a generator of (stac, uri, relative_uri) tuples
-    datasets_uris = get_items(search)
-
     # Do the indexing of all the things
     success = 0
     failure = 0
 
-    for dataset, uri in datasets_uris:
-        try:
-            index_update_dataset(
-                dataset,
-                uri,
+    sys.stdout.write("\rIndexing from STAC API...\n")
```

Comment:
Why `sys.stdout.write()` instead of logging?

Reply:
I believe logging.info disappears on some runtimes. I would love better logging levels support / structlog in these tools.

Reply:
Logging with info means datacube indexing is noisy (indexing dataset xxx for every dataset) and doing it this way (with `\r`) means that we get a fast ticking single line output, which is nice.

Reply:
Okay, that does sound nice for interactive use, so long as it doesn't mask failures.

It would be nice to have comments starting that's what's happening. I don't know \r of the top of my head.

I also agree with Tish that a better logging/setup wouldn't go astray, with different configurations for interactive or automated deployments.